lib/client-api/src/story_store.ts

```diff
@@ -145,9 +145,19 @@ export default class StoryStore extends EventEmitter {
       data === undefined ? this._selection : { storyId: data.storyId, viewMode: data.viewMode };
     this._error = error === undefined ? this._error : error;
 
+    let failed = false;
+    try {
+      if (this._channel) {
+        this._channel.emit(Events.STORY_RENDER);
+      }
+      failed = true;
```

Comment:
Should this line be `failed = false`?

I don't really understand what "failed" means here. How do we know it's failed? Do we think there may not yet be a listener on the channel for the `emit` event?

If so I reckon there are better ways to deal with that. I don't see how this code would know that.

Reply:
WDYT of this?

Reply:
I'm not sure.

1. How do we expect it to fail? Will `this._channel` be undefined? Will `.emit()` throw somehow? Or will just no-one handle the event?

2. In which case, is a `try/catch` enough to detect it?

Reply:
this._channel might be undefined

Reply:
I think that's the case in RN

Reply:
So maybe put the `isStarted=true` inside the if block then?

I want to do some more playing with the loading behaviour of SB before taking this further but thanks for taking a look.

Reply:
updated the code, WDYT?